### PR TITLE
Merged the fill_demo figures and changed the axes

### PR DESCRIPTION
**Pull request by jiyunshin**
Improvement of gallery issue #7956 
Merged fill_demo figure and fill_demo_features figure.
Modified the x-label and y-scaling.

### CODE REVIEW

**Comment by daveh19** [comment]
My students did this pull request. The continuous integration failure appears to have nothing to do with their changes. They just wrote some examples for the gallery.

File: examples/lines_bars_and_markers/fill_demo.py [hunk 1]
```diff
@@ -14,18 +14,28 @@
...
 import matplotlib.pyplot as plt
 
-x = np.linspace(0, 1, 500)
-y = np.sin(4 * np.pi * x) * np.exp(-5 * x)
+x1 = np.linspace(0, 2 * np.pi, 500)
+y1 = np.sin(x1)
+y2 = np.sin(3 * x1)
 
-fig, ax = plt.subplots()
+fig, (ax1, ax3) = plt.subplots(1,2, sharey=True)
```
Comment (phobson) [inline examples/lines_bars_and_markers/fill_demo.py]: Can we just call it `ax2`?

File: examples/lines_bars_and_markers/fill_demo.py [hunk 2]
```diff
@@ -14,18 +14,28 @@
...
 
-ax.fill(x, y, zorder=10)
-ax.grid(True, zorder=5)
+ax1.set_xticks([0, np.pi/2, np.pi, 3*np.pi/2, 2*np.pi])
+ax1.set_xticklabels( ['$0$', r'$\frac{\pi}{2}$', r'$\pi$',r'$\frac{3\pi}{2}$',r'$2\pi$'] )
```
Comment (phobson) [inline examples/lines_bars_and_markers/fill_demo.py]: remove whitespace between `(` and `[` (same for closing sides)

File: examples/lines_bars_and_markers/fill_demo.py [hunk 3]
```diff
@@ -14,18 +14,28 @@
 import numpy as np
 import matplotlib.pyplot as plt
 
-x = np.linspace(0, 1, 500)
-y = np.sin(4 * np.pi * x) * np.exp(-5 * x)
+x1 = np.linspace(0, 2 * np.pi, 500)
```
Comment (phobson) [inline examples/lines_bars_and_markers/fill_demo.py]: let's make this `x = np.linspace(0, 1, 500)`

File: examples/lines_bars_and_markers/fill_demo.py [hunk 4]
```diff
@@ -14,18 +14,28 @@
...
+ax1.fill(x1, y1, 'b', x1, y2, 'r', alpha=0.3)
+ax1.xaxis.set_label_coords(0.5, -0.1)
 
-x = np.linspace(0, 2 * np.pi, 500)
-y1 = np.sin(x)
-y2 = np.sin(3 * x)
 
-fig, ax = plt.subplots()
-ax.fill(x, y1, 'b', x, y2, 'r', alpha=0.3)
+x3 = np.linspace(0, 1, 500)
```
Comment (phobson) [inline examples/lines_bars_and_markers/fill_demo.py]: If we redefine `x1` as i described above, we can remove this

File: examples/lines_bars_and_markers/fill_demo.py [hunk 5]
```diff
@@ -14,18 +14,28 @@
...
+y3 = np.sin(4 * np.pi * x3) * np.exp(-5 * x3)
+
+ax3.set_title('fill_demo')
+ax3.set_xlabel('Time', labelpad=8)
+ax3.fill(x3, y3, zorder=10)
```
Comment (phobson) [inline examples/lines_bars_and_markers/fill_demo.py]: use `x`, `ax2`, etc

**Comment by QuLogic** [comment]
The title says "merged", but I don't see an old file being deleted?

(This would also mean 2.1, in my mind.)

**Comment by jiyunshin** [comment]
I left x1 since x = np.linspace(0, 1, 500) changes the original shape of the figure.

**Comment by phobson** [comment]
@jiyunshin Ah. you're right. I had to order of the args conflated. 

Going back to @QuLogic 's comment, is there another example that should be removed since you're merging to examples?

**Comment by jiyunshin** [comment]
We merged fill_demo.py and fill_demo_features.py into fill_demo.py.
I don't find fill_demo_features.py file in the code folder. Maybe someone already deleted?

**Comment by phobson** [comment]
I'm a little confused. Where did you originally find the example to merge these two together?

**Comment by NelleV** [comment]
It might be good to wait until #8247 is merged before doing further changes on the gallery. Some of these files have been renamed in this PR.

I wouldn't usually hold off PRs to wait for one to be merged, but this particular one has been ready to be merged for a while, and gets regular conflicts with master.

**Comment by dstansby** [comment]
http://matplotlib.org/examples/lines_bars_and_markers/fill_demo_features.html is the example that seems to have disappeared in the `master` branch; I'm happy putting it back in here though.

This just needs a PEP8 clean and a rebase; let us know @jiyunshin if you would like someone else to do that, and thanks for the PR.

**Comment by dstansby** [comment]
It looks like these have been merged by someone else: http://matplotlib.org/devdocs/gallery/lines_bars_and_markers/fill.html#sphx-glr-gallery-lines-bars-and-markers-fill-py

Thanks a lot for the PR though, and apologies I didn't get around to fixing it up before someone else made the change!

**Pull request closed without merging**